app/services/edit-agendaitem-or-subcase.js

```diff
@@ -0,0 +1,91 @@
+import Service from '@ember/service';
+import { computed } from '@ember/object';
+import { inject } from '@ember/service';
+import moment from 'moment';
+import CONFIG from 'fe-redpencil/utils/config';
+
+export default class EditAgendaitemOrSubcaseService extends Service {
+  store = inject();
+
+  setNotYetFormallyOk(itemToSet) {
+    if (itemToSet.get('formallyOk') != CONFIG.notYetFormallyOk) {
+      itemToSet.set('formallyOk', CONFIG.notYetFormallyOk);
+    }
+  }
+
+  async setNewPropertiesToModel(model, propertiesToSet, resetFormallyOk = true) {
+    if (resetFormallyOk) {
+      this.setNotYetFormallyOk(model);
+    }
+    const keys = Object.keys(propertiesToSet);
+    keys.forEach(async function (key) {
+      await model.get(key);
+      model.set(key, propertiesToSet[key]);
+    });
+
+    return model.save().then((item) => {
+      item.reload();
+      return true;
+    }).catch((e) => {
+      throw(e);
+    });
+  }
+
+  async setModifiedOnAgendaOfAgendaitem(agendaitem) {
+    const agenda = await agendaitem.get('agenda');
+    if (agenda) {
+      agenda.set('modified', moment().utc().toDate());
+      agenda.save();
+    }
+  }
+
+  getCachedProperty(property) {
+    return computed(`item.${property}`, {
+      get() {
+        const { item } = this;
+        if (item) {
+          return item.get(property);
+        } else {
+          return;
+        }
+      },
+      set: function (key, value) {
+        const { item } = this;
+        if (item) {
+          this.item.set(property, value);
+        }
+        return value;
+      }
+    })
+  }
```

Comment:
Ook zo eentje die vaak voorkomt in de codebase, maar ik snap niet goed waar het eigenlijk voor dient ... "gewoon" om `getCachedProperty('myprop')` ipv `item.myprop` te doen, of? Er zit waarschijnlijk meer achter ... die properties wrappen als een computed lijkt een soort workaround, maar ik begrijp niet goed waarvoor ...

Reply:
die method wordt in de huidige mixin gebruikt dus heb ik die ook overgenomen in de service.
Echter heb ik die niet nodig gehad in mijn ticket.
Die getCachedProp wordt op een deel componenten gebruikt, als we dit eruit willen dan pakken we dit best mee met de refactor van de hele mixin ?

Reply:
@ValenberghsSven Dat lijkt me een goed idee. die getCachedProperty wordt inderdaad op veel plaatsen gebruikt.
@MikiDi je zegt dat dat vaak voorkomt in de codebase: bedoel je de construct met get/set of "item.${property}"  of het gebruik er van? 
computed(`item.${property}` kan ik maar een keer vinden?

Reply:
This will be further expanded on in the mixin refactor, where some changes to this are planned.

Reply:
@woutersf Het gebruik er van ... Mijn doel hier was vooral van (met oog op refactoring) proberen te weten te komen "wat doet dit", want ik snap 't niet.
We willen dit inderdaaad in een andere scope oplossen waarschijnlijk, maar nu gaat het ons als we "mixin refactor" bekijken ontglippen hé ... deze change zet het al over naar een service?